src/libsystemd/sd-bus/bus-track.c

```diff
@@ -190,14 +190,8 @@ _public_ int sd_bus_track_add_name(sd_bus_track *track, const char *name) {
 
         i = hashmap_get(track->names, name);
         if (i) {
-                if (track->recursive) {
-                        unsigned k = i->n_ref + 1;
-
-                        if (k < i->n_ref) /* Check for overflow */
-                                return -EOVERFLOW;
```

Comment:
I'm not sure `track_item_ref` can replace it. As far as I can see there instead of returning `-EOVERFLOW` the counter either overflows if `systemd` is built with `NDEBUG` or the functions crashes when `assert` is triggered.

Reply:
Actually, this changes the previous behavior. But, I do not think we need to keep the previous behavior.

E.g.
```
for (unsigned i = 0; i < UINT_MAX; i++)
    sd_bus_message_ref(m);
assert_se(m->n_ref == 0);
sd_bus_message_unref(m);
```
Then, the last `sd_bus_message_unref()` triggers assertion.

Of course, in the macro which defines _ref(), we can use `assert_return(p->n_ref < UINT_MAX, NULL);`, and in bus-track.c
```
if (track->recursive && !track_item_ref(i))
        return -EOVERFLOW;
```

However, in most places, we assume `xxx_ref()` returns the same as the input. If we use `assert_return()`, then the assumption is broken...